### ISSUE TITLE
Problems using preload to get scheduler address

### ISSUE THREAD

**Issue by ntabris** [rapidsai/dask-cuda#907]
Normally we use `Nanny` class with a preload that gets the scheduler address (so that we can start EC2 instances for scheduler and workers at the same time, rather than waiting till scheduler EC2 instance has IP address before starting workers).

There are two issues I'm running into:

1. dask-cuda requires the scheduler address to be set: https://github.com/rapidsai/dask-cuda/blob/main/dask_cuda/cuda_worker.py#L112-L120
2. when dask-cuda instantiates `Nanny` it passes `preload` rather than `preload_nanny` kwarg. As far as I can tell, the `preload` kwarg on `Nanny` class doesn't do anything, it's stored as `self.preload` but then `self.preload` is set based on just `preload_nanny`. Relevant code: https://github.com/dask/distributed/blob/main/distributed/nanny.py#L162-L176

I've tested tweaks to not require scheduler address and to use `preload_nanny` kwarg, and I was able to start a cluster and have the workers successfully connect to scheduler using address that the preload fetches. Here's my fork with the tweaks:

https://github.com/rapidsai/dask-cuda/compare/main...ntabris:main?expand=1

**Comment by jrbourbeau** [comment]
> As far as I can tell, the preload kwarg on Nanny class doesn't do anything

This may not fully address this issue, but I wanted to point out that the `preload_nanny=` keyword for the `Nanny` class is used to specify the preload scripts to run on the nanny during it's startup process, while the `preload=` keyword argument to the `Nanny` class is used to specify preload scripts to run on the worker process that the nanny launches. Note when looking at the `Nanny` source code there is `Nanny.preload` for the worker preloads and `Nanny.preloads` (note the `s`) for the nanny preloads.

**Comment by ntabris** [comment]
> Note when looking at the Nanny source code there is Nanny.preload for the worker preloads and Nanny.preloads (note the s) for the nanny preloads.

Got it, I missed the `s`! Thanks for the clarification!

I think that means the issue is that dask-cuda passes preloads to the worker and doesn't have a way to pass nanny preloads, and that's what (I think) we need. Perhaps the best solution is for dask-cuda take both `preload` and `preload_nanny` kwargs (+ `argv`s)?

**Comment by jrbourbeau** [comment]
> Perhaps the best solution is for dask-cuda take both preload and preload_nanny kwargs (+ argvs)?

Yeah, that seems like it would be consistent with what the `distributed.Nanny` class is already doing upstream. @pentschev @madsbk thoughts on if adding `preload_nanny` / `preload_nanny_argv` keywords to `CUDAWorker` seems like a sensible addition? @ntabris is this something you're interested in working on?

**Comment by madsbk** [comment]
>  @pentschev @madsbk thoughts on if adding preload_nanny / preload_nanny_argv keywords to CUDAWorker seems like a sensible addition?

Yes, sounds like the right approach.

**Comment by pentschev** [comment]
> > @pentschev @madsbk thoughts on if adding preload_nanny / preload_nanny_argv keywords to CUDAWorker seems like a sensible addition?
> 
> Yes, sounds like the right approach.

Agreed, this looks like a reasonable solution. Alternatively, it seems possible to just use `"distributed.nanny.preload"` and `"distributed.nanny.preload-argv"` Dask configs as well.

**Comment by quasiben** [comment]
@ntabris are you able to move forward ?

**Comment by ntabris** [comment]
The issues aren't resolved, `dask-cuda` just hasn't been a priority at the moment and it sounds like I'll probably need to submit a PR if I want these things changed/fixed.

**Comment by quasiben** [comment]
Still an issue
